docs/advanced/additional-requests.rst

```diff
@@ -695,30 +715,39 @@ For this example, let's improve the code snippet from the previous subsection na
 
 Handling exceptions using :meth:`~.HttpClient.batch_execute` remains largely the same.
 However, the main difference is that you might be wasting perfectly good responses just
-because a single request from the batch ruined it.
+because a single request from the batch ruined it. Notice that we're using the base
+exception class of :class:`web_poet.exceptions.http.HttpError` to account for any
```

Comment:
we can also add a subclass like BatchHttpError for this case

Reply:
> or would BatchHttpError only be raised when return_exceptions=False.

I was thinking it should work like this - only raise it when return_exceptions is False.

But actually I think I misread the code, and BatchHttpError may not be a good idea :) 

So, currently we're raising HttpRequestError or HttpResponseError, and catching HttpError, to account for both situations. This sounds like a right behavior to be, better than raising BatchHttpError and catching BatchHttpError.